AutoRest/Generators/NodeJS/NodeJS/TemplateModels/MethodTemplateModel.cs

```diff
@@ -102,6 +132,19 @@ public string MethodParameterDeclarationWithCallback
         }
 
         /// <summary>
+        /// Generate the method parameter declarations with callback for a method, using TypeScript method syntax
+        /// </summary>
+        public string MethodParameterDeclarationWithCallbackTS {
+            get
+            {
+                var parameters = MethodParameterDeclarationTS;
+                var returnTypeTSString = ReturnType == null ? "void" : ReturnType.TSType(false);
+                parameters += ", callback: (err: Error, result: HttpOperationResponse<" + returnTypeTSString + ">) => void";
```

Comment:
the result will never be of type HTTPOperationResponse any more. It will be dynamic, based on the return type defined for an operation.